lib/widgets/app.dart

```diff
@@ -305,6 +310,19 @@ class _PreventEmptyStack extends NavigatorObserver {
   }
 }
 
+class _UpdateLastVisitedAccount extends NavigatorObserver {
+  _UpdateLastVisitedAccount(this.globalSettings);
+
+  final GlobalSettingsStore globalSettings;
+
+  @override
+  void didChangeTop(Route<void> topRoute, _) {
+    if (topRoute case AccountPageRouteMixin(:var accountId)) {
+      globalSettings.setInt(IntGlobalSetting.lastVisitedAccountId, accountId);
```

Comment:
Then I guess to parallel that `lastVisitedAccount` method, a corresponding mutator method:
```suggestion
      globalStore.setLastVisitedAccount(accountId);
```

That way nothing outside the implementation of GlobalStore and GlobalSettings, and the tests of GlobalSettings, needs to know the details of `.setInt(IntGlobalSetting.lastVisitedAccountId, …)` at all.